olm/src/jsMain/kotlin/io/github/matrixkt/olm/Session.kt

```diff
@@ -0,0 +1,187 @@
+        require(oneTimeKeyMsg.isNotBlank())
+        return ptr.matches_inbound(oneTimeKeyMsg)
+    }
+
+    /**
+     * Checks if the PRE_KEY([Message.MESSAGE_TYPE_PRE_KEY]) message is for this in-bound session based on the sender identity key.<br>
+     * This API may be used to process a "m.room.encrypted" event when type = 1 (PRE_KEY).
+     *
+     * @param theirIdentityKey the sender identity key
+     * @param oneTimeKeyMsg PRE KEY message
+     * @return this if operation succeed, null otherwise
+     */
+    public actual fun matchesInboundSessionFrom(theirIdentityKey: String, oneTimeKeyMsg: String): Boolean {
+        return ptr.matches_inbound_from(theirIdentityKey, oneTimeKeyMsg)
+    }
+
+    /**
+     * Encrypt a message using the session.
+     *
+     * The encrypted message is returned in a Message object.
+     *
+     * @param clearMsg message to encrypted
+     * @return the encrypted message
+     */
+    public actual fun encrypt(clearMsg: String, random: Random): Message {
+        var message = ptr.encrypt(clearMsg)
+        return Message(message.body, message.type.toLong())
+    }
+
+    /**
+     * Decrypt a message using the session.
+     * The encrypted message is given as a [Message] object.
+     * @param encryptedMsg message to decrypt
+     * @return the decrypted message
+     */
+    public actual fun decrypt(encryptedMsg: Message): String {
+        // olm_decrypt_max_plaintext_length and olm_decrypt destroy the input buffer
+        // Hence the need for two `withNativeRead`s. Should optimize this later.
+        return ptr.decrypt(encryptedMsg.type.toInt(), encryptedMsg.cipherText)
+    }
+
+    /**
+     * Return a session as a bytes buffer.
+     *
+     * The account is serialized and encrypted with [key].
+     *
+     * @param[key] encryption key
+     * @return the session as bytes buffer
+     */
+    public actual fun pickle(key: ByteArray): String {
+        return ptr.pickle(key)
+    }
+
+    public actual companion object {
+
+        /**
+         * Creates a new out-bound session for sending messages to a recipient
+         * identified by an identity key and a one time key.
+         *
+         * @param account the account to associate with this session
+         * @param theirIdentityKey the identity key of the recipient
+         * @param theirOneTimeKey the one time key of the recipient
+         */
+        public actual fun createOutboundSession(account: Account, theirIdentityKey: String, theirOneTimeKey: String, random: Random): Session {
+            require(theirIdentityKey.isNotBlank())
+            require(theirOneTimeKey.isNotBlank())
+            val jsSession = JsOlm.Session()
+            try {
+                jsSession.create_outbound(account.ptr, theirIdentityKey, theirOneTimeKey)
+                return Session(jsSession)
+            } catch (e: Exception) {
+                jsSession.free()
+                throw e
+            }
```

Comment:
It's clearer if the `try` is kept as small as possible.
```suggestion
            } catch (e: Exception) {
                jsSession.free()
                throw e
            }
            return Session(jsSession)
```